app.py

```diff
@@ -0,0 +1,61 @@
+from flask import Flask, render_template, redirect, url_for, request, session, flash
+from functools import wraps
+
+app = Flask(__name__)
+
+app.secret_key = "my precious" 
+
+def login_required(f):
+    @wraps(f)
+    def wrap(*args, **kwargs):
+        if 'logged_in' in session:
```

Comment:
Does this actually work? 
What does a sample `session` array look like? What does it contain?

Reply:
It works for logging in and grants a session code. I don't know about a session array yet, that's something I'll need to look into. I also don't have a way to log out yet so I'll need to add that in.

Reply:
👍